autometa/taxonomy/ncbi.py

```diff
@@ -364,27 +332,66 @@ def parse_merged(self):
             logger.debug('merged loaded')
         return merged
 
-    def is_common_ancestor(self, parent_taxid, child_taxid):
-        """Determines whether the provided taxids have a non-root common ancestor
+    def is_common_ancestor(self, taxid1, taxid2):
+        """
+        Determines whether the provided taxids have a non-root common ancestor
 
         Parameters
         ----------
-        parent_taxid : int
-            Description of parameter `parent_taxid`.
-        child_taxid : int
-            Description of parameter `child_taxid`.
+        taxid1 : int
+            identifer for a taxon in the Taxonomy Database by NCBI
+        taxid2 : int
+            identifer for a taxon in the Taxonomy Database by NCBI
 
         Returns
         -------
         boolean
-            True if taxids share a common ancestor and False otherwise
+            True if taxids share a common ancestor else False
         """
-        ancestor_taxid = child_taxid
-        while ancestor_taxid != 1:
-            if parent_taxid == ancestor_taxid:
-                return True
-            ancestor_taxid = self.parent(ancestor_taxid)
-        return False
+        taxid1_lineage = self.lineage(taxid1)
+        taxid2_lineage = self.lineage(taxid2)
+        taxid1_lineage_taxids = set([taxids.get('taxid')
+                                     for taxids in taxid1_lineage])
+        taxid2_lineage_taxids = set([taxids.get('taxid')
+                                     for taxids in taxid2_lineage])
+        print(taxid1_lineage_taxids)
+        print(taxid2_lineage_taxids)
+        common_ancestor = taxid2_lineage_taxids & taxid1_lineage_taxids
+        print(common_ancestor)
+        return True if common_ancestor else False
+
+    def is_valid_taxid(self, taxid):
+        """
+        Checks if the given taxid is a positive interger or not and is present in either 
+        'names.dmp' or 'merged.dmp'
+
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in the Taxonomy Database by NCBI
+
+        Returns
+        -------
+        boolean 
+            True if the 'taxid' is a positive interger and is present in either 'names.dmp' or 'merged.dmp' else False 
+        """
+        try:
+            taxid = float(taxid)
+        except ValueError:
```

Comment:
I thought we wanted to remove a try and except statement like this and replace it with something like `isinstance`?

Reply:
I tried different ways but then none of them were able to resolve the case when a number is entered as a string. eg. `"526"`. Is there any particular reason for not using a `try and except` block?

Reply:
The problem here is that, for example `"inf"` can be cast as a float. Also, non-integer floats are not valid taxids! You could do:

```python
if isinstance(taxid, str):
    # Check if it is an int
    if taxid.isnumeric():
        taxid = int(taxid)
    # Check if it is a float
    if taxid.replace('.', '', 1).isnumeric():
        if float(taxid).is_integer():
            taxid = int(taxid)
        else:
            # The taxid is a non-integer float, so you should do some handling here
    else:
        # Taxid is an invalid type of string at this point

```